### DISCUSSION TITLE
How to always repaint

### DISCUSSION THREAD

**Discussion by dejang** [emilk/egui#4721] [Q&A]
I am developing a music player with egui and while playing a song the UI only updates while I move the mouse or do some kind of interaction. I'd like it to always update. Is that possible with egui ?

**Comment by YgorSouza** [comment]
Yes, you just have to make sure [Context::request_repaint](https://docs.rs/egui/0.27.2/egui/struct.Context.html#method.request_repaint) gets called on every frame. There is an example in the demo app:

https://github.com/emilk/egui/blob/a0f4fafb88a71b8cc08921259344c3574c8ba880/crates/egui_demo_app/src/backend_panel.rs#L61-L74

**Reply by dejang** [reply]
Ahh thanks, basically calling ctx.request_repaing() in the `update` method of the App repaints the whole thing. This is what I needed.
